charts/Controls.tsx

```diff
@@ -105,6 +104,36 @@ class ShareMenu extends React.Component<{ chart: ChartConfig, chartView: any, on
         document.removeEventListener('click', this.onClickSomewhere)
     }
 
+    @observable showCopyResulltText: boolean = false
+
+    @action.bound onCopy() {
+        if (!this.canonicalUrl) { return }
+
+        // Adapted from https://github.com/30-seconds/30-seconds-of-code#copytoclipboard-
+        const copyToClipboard = (str: string) => {
+            const el = document.createElement('textarea')
+            el.value = str
+            el.setAttribute('readonly', '')
+            el.style.position = 'absolute'
+            el.style.left = '-9999px'
+            document.body.appendChild(el)
+            const selected =
+                document.getSelection().rangeCount > 0 ? document.getSelection().getRangeAt(0) : false
```

Comment:
The type checker complains that `document.getSelection()` can be `null` here, it needs a `!== null` check before we try to access properties.

To run the type checker execute `yarn typecheck`